src/borg/repository.py

```diff
@@ -463,6 +463,10 @@ def open(self, path, exclusive, lock_wait=None, lock=True):
         self.id = unhexlify(self.config.get("repository", "id").strip())
         self.io = LoggedIO(self.path, self.max_segment_size, self.segments_per_dir)
 
+    def reopen(self, exclusive=True):
```

Comment:
The original test suite has a reopen method that created a new repository object between the `__exit__` and `__enter__` 

I didn't think it would be correct to do the same thing here, although I was not sure how to mimc this behavior, and thus all the tests that call this reopen function are having issues. I spent  _several_  hours trying get reopen to work as it does in the original test file, to no success.

Reply:
Instead of reopen, have a fixture that's returns a non-reentreant context manager which yields the repo on entering

Reply:
I will give this a try - thank you :)